test/general/openpgp.js

```diff
@@ -2588,6 +2703,100 @@ amnR6g==
       });
 
     });
+
+    describe('Specific encryption/signing key testing', async function () {
+      const encryptionKeyIds = [
+        keyIdType.fromId("87EAE0977B2185EA"),
+        keyIdType.fromId("F94F9B34AF93FA14"),
+        keyIdType.fromId("08F7D4C7C59545C0")
+      ];
+      const signingKeyIds = [
+        keyIdType.fromId("663277AF60400638"),
+        keyIdType.fromId("BBE14491E6EE6366"),
+        keyIdType.fromId("3E0F20F1A71D6DFD")
+      ];
+      const getPrimaryKey = async () => openpgp.readArmoredKey(
+        multipleEncryptionAndSigningSubkeys
+      );
+
+      it('Encrypt message with a specific encryption key id', async function () {
+        const primaryKey = await getPrimaryKey();
+        let m;
+        let p;
+        for (let i = 0; i < encryptionKeyIds.length; i++) {
+          m = await openpgp.readArmoredMessage(await openpgp.encrypt({
+            message: openpgp.Message.fromText("Hello World\n"),
+            publicKeys: primaryKey,
+            encryptionKeyIds: [encryptionKeyIds[i]]
+          }));
+          p = m.packets.filterByTag(openpgp.enums.packet.publicKeyEncryptedSessionKey);
+          expect(p.length).equals(1);
+          expect(p[0].publicKeyId.equals(encryptionKeyIds[i])).to.be.true;
+        }
+      });
+
+      it('Sign message with a specific signing key id', async function () {
+        const primaryKey = await getPrimaryKey();
+        let s;
+        let p;
+        for (let i = 0; i < signingKeyIds.length; i++) {
+          s = await openpgp.readArmoredSignature(await openpgp.sign({
+            message: openpgp.Message.fromText("Hello World\n"),
+            privateKeys: primaryKey,
+            signingKeyIds: [signingKeyIds[i]],
+            detached: true
+          }));
+          p = s.packets.filterByTag(openpgp.enums.packet.signature);
+          expect(p.length).equals(1);
+          expect(p[0].issuerKeyId.equals(signingKeyIds[i])).to.be.true;
+        }
+      });
+
+      it('Encrypt and sign with specific encryption/signing key ids', async function () {
+        const primaryKey = await getPrimaryKey();
+        const latestEncryptionKeyId = (await primaryKey.getEncryptionKey()).getKeyId();
+        const latestSigningKeyId = (await primaryKey.getSigningKey()).getKeyId();
+        const plaintextMessage = openpgp.Message.fromText("Hello World\n");
+
+        const checkEncryptedPackets = (encryptionKeyIds, pKESKList) => {
+          pKESKList.forEach(({ publicKeyId }, i) => {
+            if (!encryptionKeyIds[i]) {
+              expect(publicKeyId.equals(latestEncryptionKeyId)).to.be.true;
```

Comment:
Is this line ever reached? Same for `checkSignatures`. Perhaps a 4th `primaryKey` is missing as input of encryption private & publicKeys?

Reply:
It's not reached, I'll bring it back, but the heavy test already passed successfully when the *keyId was not specified for a public/private key

Reply:
Alternatively, I think you can also just remove these (now unused) lines, as it's indeed already tested elsewhere

Reply:
Okay, I removed it